.gitlab-ci.yml

```diff
@@ -1405,69 +1407,79 @@ kitchen_windows_upgrade6-a7:
 # run dd-agent-testing on centos
 kitchen_centos_chef-a6:
   allow_failure: false
+  needs: ["deploy_rpm_testing-a6"]
   <<: *skip_when_unwanted_on_6
   <<: *kitchen_centos_common
   <<: *kitchen_a6_common
   <<: *kitchen_chef_common
 
 kitchen_centos_chef-a7:
   allow_failure: false
+  needs: ["deploy_rpm_testing-a7"]
```

Comment:
Nit: we can refactor this so that the `needs` only need to be defined once per platform x Agent version combination. To do so:
- create an intermediate job for the platform x Agent version:

```yaml
.kitchen_<platform>_<agent_version>_common: &kitchen_<platform>_<agent_version>_common
  <<: *kitchen_<platform>_common
  <<: *kitchen_<agent_version>_common
  needs: ["deploy_<package_type>_testing-<agent_version>"]
```

- change the final kitchen jobs to:
```yaml
kitchen_<platform>_<test_suite>-<agent_version>:
  allow_failure: false
  <<: *kitchen_<platform>_<agent_version>_common
  <<: *kitchen_<test_suite>_common
```

We're already doing this for Windows (because we needed a slightly different structure for the Windows kitchen jobs), so we could implement it for Linux now that there's a reason to do so.

Reply:
We could then also move the `skip_when_unwanted_on_<agent_version>` to the `.kitchen_<platform>_<agent_version>_common`. Or do you think it's more clear to have it on every entry?

Reply:
I've been thinking about this, and actually I think`<<: *skip_when_unwanted_on_<other_agent_version>` can be moved to `.kitchen_<agent_version>_common` directly

Reply:
Yeah, even better.